app/models/ems_refresh/save_inventory_infra.rb

```diff
@@ -287,6 +288,12 @@ def save_resource_pools_inventory(ems, hashes, target = nil)
 
   def save_storage_profiles_inventory(ems, hashes, _target = nil)
     save_inventory_multi(ems.storage_profiles, hashes, :use_association, [:ems_ref])
+
+    # Collect the ids of storage_profiles after saving
```

Comment:
`store_ids_for_new_records` might do what you need here, might be able to use `save_inventory_assoc`